### ISSUE TITLE
DataTable: ClassCastException when DataTable is lazy and value is null

### ISSUE THREAD

**Issue by brainlag** [primefaces/primefaces#10371]
### Describe the bug

I get a ClassCastException with Primefaces 13 when the value of a lazy DataTable is null.

When the value is null `getDataModel()` returns an empty `ListDataModel` which is not a lazy model type.

This worked with Primefaces 12 and before. 

```
java.lang.ClassCastException: class javax.faces.model.ListDataModel cannot be cast to class org.primefaces.model.LazyDataModel (javax.faces.model.ListDataModel and org.primefaces.model.LazyDataModel are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @1687eb01)
	at org.primefaces.component.datatable.DataTable.getLazyDataModel(DataTable.java:1209) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.component.datatable.DataTable.loadLazyDataIfEnabled(DataTable.java:477) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.component.datatable.DataTableRenderer.preRender(DataTableRenderer.java:126) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.component.datatable.DataTableRenderer.render(DataTableRenderer.java:103) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.component.datatable.DataTableRenderer.encodeEnd(DataTableRenderer.java:96) ~[primefaces-13.0.0.jar:13.0.0]
	at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:600) ~[jakarta.faces-2.3.18.jar:2.3.18]
	at org.primefaces.renderkit.CoreRenderer.renderChild(CoreRenderer.java:85) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.renderkit.CoreRenderer.renderChildren(CoreRenderer.java:67) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.component.panel.PanelRenderer.encodeContent(PanelRenderer.java:235) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.component.panel.PanelRenderer.encodeMarkup(PanelRenderer.java:144) ~[primefaces-13.0.0.jar:13.0.0]
	at org.primefaces.component.panel.PanelRenderer.encodeEnd(PanelRenderer.java:77) ~[primefaces-13.0.0.jar:13.0.0]
	at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:600) ~[jakarta.faces-2.3.18.jar:2.3.18]
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1655) ~[jakarta.faces-2.3.18.jar:2.3.18]
	at javax.faces.render.Renderer.encodeChildren(Renderer.java:152) ~[jakarta.faces-2.3.18.jar:2.3.18]
	at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:571) ~[jakarta.faces-2.3.18.jar:2.3.18]
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1648) ~[jakarta.faces-2.3.18.jar:2.3.18]
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1651) ~[jakarta.faces-2.3.18.jar:2.3.18]
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1651) ~[jakarta.faces-2.3.18.jar:2.3.18]
```

### PrimeFaces edition

Community

### PrimeFaces version

13.0.0

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

17

**Comment by tandraschko** [comment]
please provide a reproducer

**Comment by brainlag** [comment]
It is as simple as this:

```
<p:dataTable lazy="true" value="#{testBean.data}"/>
```
With following bean:
```
import javax.faces.view.ViewScoped;
import javax.inject.Named;

import org.primefaces.model.LazyDataModel;

@Named
@ViewScoped
public class TestBean implements Serializable {
	private static final long serialVersionUID = 3916728926095954508L;

	public LazyDataModel getData() {
		return null;
	}
}
```

**Comment by Rapster** [comment]
I see, but I don't think there is much to fix here except your scenario. It is by design that whenever value is null then `UIData#getDataModel` will return a `ListDataModel` by default. If you can elaborate your scenario that might explain why your lazydatamodel is null, maybe we can do something about it

**Comment by tandraschko** [comment]
i dont think its a bug but i think it would be a good enhancement to support this

**Comment by melloware** [comment]
Most likely related to this: https://github.com/primefaces/primefaces/issues/8436

**Comment by Rapster** [comment]
i'll see if I can do something about it

**Comment by tandraschko** [comment]
this fixes the problem, please apply if you are ok with it @Rapster:

```
    public LazyDataModel<?> getLazyDataModel() {
        if (isLazy()) {
            Object value = getValue();
            if (value == null) {
                return null;
            }
            return (LazyDataModel<?>) getDataModel();
        }
        return null;
    }
```

**Comment by ikitzmann** [comment]
@tandraschko is it possible to get the last fix also in the PF 13 version?

**Comment by tandraschko** [comment]
Check the assigned milestone Here ;)

**Comment by ikitzmann** [comment]
@tandraschko Is this true for https://github.com/primefaces/primefaces/commit/311148d8cbce32a6c1a998c9a978184dc4593b5b ? This seems only to be in the master branch

**Comment by tandraschko** [comment]
Did you try 13.0.1?

**Comment by ikitzmann** [comment]
@tandraschko 
this:
![image](https://github.com/primefaces/primefaces/assets/81322062/37b89848-d95b-4e39-84ce-30768b04edf1)
is not visible in 13.X branch:
https://github.com/primefaces/primefaces/blob/13.X/primefaces/src/main/java/org/primefaces/component/datatable/DataTable.java

We currently use 13.0.2

**Comment by melloware** [comment]
@tandraschko it looks like its marked 13.0.1 but never got into 13.0.1.  I can add it.

**Comment by tandraschko** [comment]
https://github.com/primefaces/primefaces/blob/13.X/primefaces/src/main/java/org/primefaces/component/datatable/DataTable.java#L1211

This should already be enough, it was just refactored later

Therefor i asked if you have this issue in 13.0.1

**Comment by ikitzmann** [comment]
We did test both 13.0.1 and 13.0.2 and have a similar ClassCastException with a lazy=true DataTable.
Therefore we did look at the bug issues and found this one.

It might be a different/related bug.

**Comment by tandraschko** [comment]
can you try with the PR then?
otherwise we need a reproducer

**Comment by ikitzmann** [comment]
@tandraschko We have now made our own DataTable Subclass which only overwrites the getLazyDataModel() method with the content of https://github.com/melloware/primefaces/blob/58b4b4119a1672db74740b6666471da313c5149c/primefaces/src/main/java/org/primefaces/component/datatable/DataTable.java

After activating it in faces-config.xml the exception is gone.

**Comment by fernanvergara** [comment]
Hi, was this bug fixed in PF 13.x?

**Comment by melloware** [comment]
It was fixed please use 13.0.10
